File: farthestNode.cpp
#include <bits/stdc++.h>
using namespace std;

vector<vector<int>> v(20001);
vector<int> ans(20001);
int check[20001];

int solution(int n, vector<vector<int>> edge) {
    for (int i = 0; i < edge.size(); i++) {
        v[edge[i][0]].push_back(edge[i][1]);
        v[edge[i][1]].push_back(edge[i][0]);
    }
    int max_level = 0;
    queue<pair<int, int>> q;
    q.push({1, 0});
    check[1] = 1;
    while (!q.empty()) {
        int x = q.front().first;
        int level = q.front().second;
        q.pop();
        if (max_level < level) max_level = level;
        ans[level]++;
        for (int i = 0; i < v[x].size(); i++) {
            int nx = v[x][i];
            if (check[nx] == 0) {
                check[nx] = 1;
                q.push({nx, level + 1});
            }
        }
    }
    return ans[max_level];
}

File: farthestNode.js
function solution(n, edge) {
    let a = [];
    let check = [];
    for (let i = 0; i <= n; i++) {
        a[i] = [];
        check[i] = -1;
    }
    for (let i = 0; i < edge.length; i++) {
        a[edge[i][0]].push(edge[i][1]);
        a[edge[i][1]].push(edge[i][0]);
    }
    let q = [1];
    check[1] = 0;
    let maxv = -1;
    let cnt = 1;
    while (q.length != 0) {
        let x = q[0];
        q.shift();
        if (maxv == check[x]) {
            cnt++;
        }
        if (maxv < check[x]) {
            maxv = check[x];
            cnt = 1;
        }
        for (let i = 0; i < a[x].length; i++) {
            let nx = a[x][i];
            if (check[nx] == -1) {
                check[nx] = check[x] + 1;
                q.push(nx);
            }
        }
    }
    return cnt;
}
